core/arch/arm/plat-stm32mp1/scmi_server.c

```diff
@@ -812,18 +812,27 @@ int32_t plat_scmi_voltd_levels_array(unsigned int channel_id,
 	}
 }
 
-long plat_scmi_voltd_get_level(unsigned int channel_id, unsigned int scmi_id)
+int32_t plat_scmi_voltd_get_level(unsigned int channel_id, unsigned int scmi_id,
+				  int32_t *level)
 {
 	struct stm32_scmi_voltd *voltd = find_voltd(channel_id, scmi_id);
+	long voltage = 0;
 
 	if (!voltd)
-		return 0;
+		return SCMI_INVALID_PARAMETERS;
 
 	switch (voltd->priv_dev) {
 	case VOLTD_PWR:
-		return pwr_get_level(voltd);
+		*level = pwr_get_level(voltd);
```

Comment:
`pwr_get_level()` returns `long`, `*level` is `int32_t`

Reply:
Let's hear from @etienne-lms but it looks like more consistency would be welcome...

Reply:
For consistency in the API functions, I think `level` should be of type `long`. `scmi_voltd_level_get()` should do the conversion to `int32`.

Reply:
Ok, I've changed level as described. The force pushes fixed checkpatch errors etc. on the fixup commit.

Reply:
Ready to merge? Then I'll flatten the commit stack.